apps/meteor/app/livechat/server/lib/contacts/disableContact.ts

```diff
@@ -0,0 +1,23 @@
+import type { ILivechatContact } from '@rocket.chat/core-typings';
+import { LivechatContacts, LivechatRooms } from '@rocket.chat/models';
+
+import { settings } from '../../../../settings/server';
+import { removeGuest } from '../guests';
+
+export async function disableContactById(contactId: string): Promise<void> {
+	const contact = await LivechatContacts.findOneEnabledById<Pick<ILivechatContact, '_id' | 'channels'>>(contactId);
+	if (!contact) {
+		throw new Error('error-contact-not-found');
+	}
+
+	// Checking if the contact has any open channel/room before removing its data.
+	const contactOpenRooms = await LivechatRooms.findOpenByContactId(contactId).toArray();
```

Comment:
You're not using the rooms for other things. You can create a new method to just count the rooms 👀

Reply:
I'd prefer using a `findOne` with `{ projection: { _id: 1 } }` though, since you don't need the total count of rooms, you just need to know if there are any